### PR TITLE
[Sparse] Support scalar dividing diagonal

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
A sub PR of #5148 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: python/dgl/sparse/elementwise_op_diag.py [hunk 1]
```diff
@@ -157,14 +158,50 @@ def diag_div(D1: DiagMatrix, D2: Union[DiagMatrix, Scalar]) -> DiagMatrix:
...
     elif is_scalar(D2):
-        assert D2 != 0, "Division by zero is not allowed."
         return diag(D1.val / D2, D1.shape)
     else:
         # Python falls back to D2.__rtruediv__(D1) then TypeError when
         # NotImplemented is returned.
         return NotImplemented
 
 
+def diag_rdiv(D1: DiagMatrix, D2: Union[DiagMatrix, Scalar]) -> DiagMatrix:
```
Comment (mufeili) [inline python/dgl/sparse/elementwise_op_diag.py:168]: Does torch sparse also support this?

Comment (mufeili) [reply python/dgl/sparse/elementwise_op_diag.py:168]: Also is this also supported for sparse matrices?

**Review by mufeili — COMMENTED** [review]
Minor comments

File: python/dgl/sparse/elementwise_op_diag.py (hunk 1 above)
Comment (BarclayII) [reply python/dgl/sparse/elementwise_op_diag.py:168]: Figured out that this is not supported by PyTorch sparse.  I'll close this PR for now.

**Comment by frozenbugs** [comment]
Let's hold this after 1.0, we should be able to define all element-wise ops combinations.

**Pull request closed without merging**